### ISSUE TITLE
ETL plugin UI needs to handle correctly "remote" database URL

### ISSUE THREAD

**Issue by MironAtHome** [orientechnologies/orientdb#8726]
## Expected behavior
First of all, thanks for making this database. Rest will be critique ( pretty much in the eye ) but it shouldn't derail the overarching sense of pleasure of working with your product. And if I do sound overbearing, please write it off to my personal shortcomings.

Now, to the first point, of what I was really expecting. I was expecting some kind of command, such as "INSERT" where I can provide "in" and "out" fields, to define the edges using select statement. Well, I couldn't find one anywhere so far. There is an "CREATE EDGE" variant that allows to specify select statement using "to" and "from", but those are two distinct, separate queries. And while ( it would be logical ) this variant allows multiple "from", it would still require to somehow constraint to a specific value for the "to". In my case there is more than a single "to" and for me such a query has to provide for the capability to iterate over set of vertices. Taking research of approach using SQL statement one step further, there is a "match" command, which allows using context variable $elements $currentMatch and $matched, in an ideal world I should be able to pass context variable of my choice from the query defined for the "from" to the query performed for the "to" edge link, however, there is no example or explanation how it works, or whether it works at all. Such a functionality would be par for the course to what an SQL-99 standard defines as "RECURSIVE" subsection under the section "QUERIES". To summarise, if I could improve something, it would be construction of documentation for the SQL statements, and the example section along the way. And, well, I could, at the moment, only dream of stored procedures, where I could using something like a "WHILE" :)
Back to the original goal of excercise, to iterate over the set of vertices I had to resort to defining "CREATE EDGE" using output of another SQL query, taking it as a script and fetching it to console application. And since the method of scripting these steps isn't clear to me ( I am total newby ) I had to revert to the only qpproach, short of writing an short application in Java or some other programming langauge, using ETL plugin. At this moment I am experimenting with its UI based implementation and while doing so ran into following behavior, please read below in the "Actual Behavior" section.

Here is expectation: I should be able to run ETL process with JDBC extractor running an query against my database, let us call it "GEO" while loader loads edges, as a result of transforms, performed by ETL. I am not ready to subscribe to having to load product of ETL to some "other" "temporary" "stage" database, from whence to load those into intended target database.

## Actual behavior  
When running ETL script error message is produced. Message reads:
[OEnterpriseLocalPaginatedStorage]File is already locked by other thread
java.nio.channels.OverlappingFileLockException

## Steps to reproduce  
Define ETL using attached JSON file. Try to run it.

## 1
Adding non - blocking load, when the following flags ( both ) not set to "true" in the method beginLoader of OETLOrientDBLoader:

--> use current approach
    if((dbAutoCreate == true) && (dbAutoDropIfExists == true)) {
        db = pool.acquire();
        db.activateOnCurrentThread();
--> else
    instead of acquiring pool use logic copied from OConsoleDatabaseApp to secure connection to target database.
## 2
consider changing loader to an selection of connection methods, add JDBC loader

## in other words, considering various non - blocking loader strategies. I think #1 is reasonable enough to consider for major . release.

**Comment by MironAtHome** [comment]
Following up on steps 1 and 2.
Modified code, run loader, same issue. After checking JSON it turns out that loading JSON with loader database URL defined as "remote:localhost/GEO" the application overwrites JSON as "plocal:<database_path>", which leaders to same issue.
Currently looking how to obviate this behavior.
Slight plan of original plan, initial intent of using boolean flags changed to using "kind" variable, defined a few lines further in same loader class. When connection URL defined as "embedded" or "plocal" keep as is, for "remote" using modified approach.

**Comment by MironAtHome** [comment]
Turns out this is just an ETL UI ussie.
oetl tool, provided as part of distribution, accepts loader with dbURL as "remote" and properly accesses database, without failing to lock file access.
Will refocus on UI changes to properly handle "remote" loader and update issue title appropriately.
For the record, here is first take how one could iterate over set of values, produced by query executed by extractor:
  "transformers": [
    {
      "command": {
        "language": "sql",
        "command": "CREATE EDGE CountryCity FROM [${input.LinkTableToCity}] TO [${input.LinkTableToCountry}];",
        "output" : "edge"
      }
    }
  ]
One doesn't need to specify output class name in the loader, when performing creation of edge in the command transformer.
Please note, this is pretty much the worst approach to take, in terms of both, speed and resource usage. So, this is just an example.
